### PR TITLE
Feat: health endpoint

### PR DESCRIPTION
**Pull request by xiomaraR**
## Description

This pull request introduces the addition of a health endpoint along with related tests. It also includes modifications to the authentication middleware so that the health endpoint is bypassed. 

## Commit type

  - `feat`: indicates the addition of a new feature or functionality to the project.
 

## Issue

Create health endpoint [#134 ](https://github.com/AntonioMrtz/SpotifyElectron/issues/134)

## Solution

The health endpoint is added to provide a health check for the application, which can be useful for monitoring purposes. The modifications to bypass authentication for the health endpoint keeps it simple and accessible.

## Proposed Changes

-Added health endpoint router and related tests
-Included health router in `__main__.py`
-Modified authentication middleware to bypass authentication for health endpoint
-Utilized Starlette's named constants for status codes

## Potential Impact

These changes are expected to have a positive impact on the project by enhancing its monitoring capabilities.

## Tests Performed

- [✅ ] Test performed 

## Additional Tasks

-Review and address any feedback received during the pull request review process

## Assigned

@AntonioMrtz

### CODE REVIEW

**Comment by xiomaraR** [comment]
This PR replaces [#139 ](https://github.com/AntonioMrtz/SpotifyElectron/pull/139)

File: Backend/app/__main__.py [hunk 1]
```diff
@@ -16,7 +16,16 @@
     allowed_origins,
     max_age,
 )
-from app.routers import artistas, canciones, generos, login, playlists, search, usuarios
+from app.routers import (
```
Comment (AntonioMrtz) [inline Backend/app/__main__.py:19]: Imports format shouldnt change. Ruff doesnt format the inline imports that appeared before.

File: Backend/app/services/user_service.py [hunk 2]
```diff
@@ -3,10 +3,10 @@
 
 from fastapi import HTTPException
 
-import app.services.security_service as security_service
 from app.database.Database import Database
 from app.model.TokenData import TokenData
 from app.model.User import User
+from app.services import security_service
```
Comment (AntonioMrtz) [inline Backend/app/services/user_service.py:9]: Revert change in imports

File: Backend/app/services/security_service.py [hunk 3]
```diff
@@ -6,15 +6,13 @@
...
-import app.services.all_users_service as all_users_service
-import app.services.artist_service as artist_service
-import app.services.user_service as user_service
 from app.boostrap.PropertiesManager import PropertiesManager
 from app.constants.set_up_constants import DISTRIBUTION_ID_ENV_NAME
 from app.model.Artist import Artist
 from app.model.TokenData import TokenData
 from app.model.User import User
 from app.model.UserType import User_Type
+from app.services import all_users_service, artist_service, user_service
```
Comment (AntonioMrtz) [inline Backend/app/services/security_service.py:15]: Revert change in imports

File: Backend/app/services/search_service.py [hunk 4]
```diff
@@ -1,8 +1,6 @@
 from fastapi import HTTPException
 
-import app.services.artist_service as artist_service
-import app.services.playlist_service as playlist_service
-import app.services.user_service as user_service
+from app.services import artist_service, playlist_service, user_service
```
Comment (AntonioMrtz) [inline Backend/app/services/search_service.py:3]: Revert change in imports

File: Backend/app/services/playlist_service.py [hunk 5]
```diff
@@ -3,11 +3,10 @@
 
 from fastapi import HTTPException
 
-import app.services.all_users_service as all_users_service
-import app.services.dto_service as dto_service
 from app.database.Database import Database
 from app.model.Playlist import Playlist
 from app.model.TokenData import TokenData
+from app.services import all_users_service, dto_service
```
Comment (AntonioMrtz) [inline Backend/app/services/playlist_service.py:9]: Revert change in imports

File: Backend/app/services/artist_service.py [hunk 6]
```diff
@@ -3,10 +3,10 @@
 
 from fastapi import HTTPException
 
-import app.services.security_service as security_service
 from app.database.Database import Database
 from app.model.Artist import Artist
 from app.model.TokenData import TokenData
+from app.services import security_service
```
Comment (AntonioMrtz) [inline Backend/app/services/artist_service.py:9]: Revert change in imports

File: Backend/app/services/all_users_service.py [hunk 7]
```diff
@@ -1,10 +1,8 @@
 from fastapi import HTTPException
 
-import app.services.artist_service as artist_service
-import app.services.playlist_service as playlist_service
-import app.services.user_service as user_service
 from app.model.TokenData import TokenData
 from app.model.UserType import User_Type
+from app.services import artist_service, playlist_service, user_service
```
Comment (AntonioMrtz) [inline Backend/app/services/all_users_service.py:5]: Revert change in imports

File: Backend/app/services/song_services/song_service_db_blob.py [hunk 8]
```diff
@@ -6,14 +6,13 @@
...
 
-import app.services.artist_service as artist_service
-import app.services.dto_service as dto_service
 from app.database.Database import Database
 from app.model.DTO.SongDTO import SongDTO
 from app.model.Genre import Genre
 from app.model.Song import Song
 from app.model.SongBlob import SongBlob
 from app.model.TokenData import TokenData
+from app.services import artist_service, dto_service
```
Comment (AntonioMrtz) [inline Backend/app/services/song_services/song_service_db_blob.py:15]: Revert change in imports

File: Backend/app/services/song_services/song_service_aws_sdk.py [hunk 9]
```diff
@@ -11,13 +11,12 @@
...
 from pymongo.errors import PyMongoError
 
-import app.services.artist_service as artist_service
-import app.services.dto_service as dto_service
 from app.database.Database import Database
 from app.model.DTO.SongDTO import SongDTO
 from app.model.Genre import Genre
 from app.model.Song import Song
 from app.model.TokenData import TokenData
+from app.services import artist_service, dto_service
```
Comment (AntonioMrtz) [inline Backend/app/services/song_services/song_service_aws_sdk.py:19]: Revert change in imports

File: Backend/app/services/song_services/song_service_aws_lambda.py [hunk 10]
```diff
@@ -7,15 +7,14 @@
...
-import app.services.artist_service as artist_service
-import app.services.dto_service as dto_service
 from app.boostrap.PropertiesManager import PropertiesManager
 from app.constants.set_up_constants import LAMBDA_URL_ENV_NAME
 from app.database.Database import Database
 from app.model.DTO.SongDTO import SongDTO
 from app.model.Genre import Genre
 from app.model.Song import Song
 from app.model.TokenData import TokenData
+from app.services import artist_service, dto_service
```
Comment (AntonioMrtz) [inline Backend/app/services/song_services/song_service_aws_lambda.py:17]: Revert change in imports

File: Backend/app/routers/usuarios.py [hunk 11]
```diff
@@ -3,9 +3,7 @@
 from fastapi import APIRouter, Body, Header, HTTPException
 from fastapi.responses import Response
 
-import app.services.all_users_service as all_users_service
-import app.services.security_service as security_service
-import app.services.user_service as user_service
+from app.services import all_users_service, security_service, user_service
```
Comment (AntonioMrtz) [inline Backend/app/routers/usuarios.py:6]: Revert change in imports

File: Backend/app/routers/search.py [hunk 12]
```diff
@@ -4,7 +4,7 @@
 from fastapi.encoders import jsonable_encoder
 from fastapi.responses import Response
 
-import app.services.search_service as search_service
+from app.services import search_service
```
Comment (AntonioMrtz) [inline Backend/app/routers/search.py:7]: Revert change in imports

File: Backend/app/routers/playlists.py [hunk 13]
```diff
@@ -5,8 +5,7 @@
 from fastapi.encoders import jsonable_encoder
 from fastapi.responses import Response
 
-import app.services.playlist_service as playlist_service
-import app.services.security_service as security_service
+from app.services import playlist_service, security_service
```
Comment (AntonioMrtz) [inline Backend/app/routers/playlists.py:8]: Revert change in imports

File: Backend/app/routers/login.py [hunk 14]
```diff
@@ -1,12 +1,12 @@
...
-from datetime import datetime, timedelta, timezone
+from datetime import UTC, datetime, timedelta
 from typing import Annotated
 
 from fastapi import APIRouter, Depends
 from fastapi.responses import Response
 from fastapi.security import OAuth2PasswordRequestForm
 
-import app.services.security_service as security_service
+from app.services import security_service
```
Comment (AntonioMrtz) [inline Backend/app/routers/login.py:9]: Revert change in imports

File: Backend/app/routers/login.py [hunk 15]
```diff
@@ -1,12 +1,12 @@
 import json
-from datetime import datetime, timedelta, timezone
+from datetime import UTC, datetime, timedelta
```
Comment (AntonioMrtz) [inline Backend/app/routers/login.py:2]: I made the exact same change some weeks ago and the app broke. In the future i will check what happened but it should be revert for now.

File: Backend/app/routers/health.py [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+@router.get("/", summary="Health Check Endpoint")
+def get_health():
+    """Health endpoint allows us to determine if the app has launched correctly
+
+    Returns
+    -------
+        Response 200 OK
+
+    """
+    return Response(status_code=200, content="OK", media_type="text/plain")
```
Comment (AntonioMrtz) [inline Backend/app/routers/health.py:16]: Using `from starlette.status import HTTP_200_OK` its recommended

File: Backend/app/routers/generos.py [hunk 17]
```diff
@@ -1,7 +1,7 @@
 from fastapi import APIRouter
 from fastapi.responses import Response
 
-import app.services.genre_service as genre_service
+from app.services import genre_service
```
Comment (AntonioMrtz) [inline Backend/app/routers/generos.py:4]: Revert change in imports

File: Backend/app/routers/canciones.py [hunk 18]
```diff
@@ -5,9 +5,8 @@
 from fastapi.encoders import jsonable_encoder
 from fastapi.responses import Response
 
-import app.services.dto_service as dto_service
-import app.services.security_service as security_service
 from app.model.Genre import Genre
+from app.services import dto_service, security_service
```
Comment (AntonioMrtz) [inline Backend/app/routers/canciones.py:9]: Revert change in imports

File: Backend/app/routers/artistas.py [hunk 19]
```diff
@@ -5,8 +5,7 @@
 from fastapi.encoders import jsonable_encoder
 from fastapi.responses import Response
 
-import app.services.artist_service as artist_service
-import app.services.security_service as security_service
+from app.services import artist_service, security_service
```
Comment (AntonioMrtz) [inline Backend/app/routers/artistas.py:8]: Revert change in imports

File: Backend/app/middleware/CheckJwtAuthMiddleware.py [hunk 20]
```diff
@@ -2,9 +2,9 @@
 from starlette.middleware.base import BaseHTTPMiddleware
 from starlette.status import HTTP_401_UNAUTHORIZED
 
-import app.services.security_service as security_service
 from app.logging.logger_constants import LOGGIN_CHECK_AUTH_JWT_MIDDLEWARE
 from app.logging.logging_schema import SpotifyElectronLogger
+from app.services import security_service
```
Comment (AntonioMrtz) [inline Backend/app/middleware/CheckJwtAuthMiddleware.py:7]: Revert change in imports

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]
Hi, first of all thanks for contributing to the project and taking the time to solve all the issues you had with the setup. 

Its recommend in PRs that the only changes included should be related to the issue. Some changes are out of scope:

* Imports re-formatting and changing alias
* Non related files changes such as .env.example, docker scripts

The overall PR is really good, the health endpoint was correct, tests were good and health endpoint was added into middleware bypass paths.

For the PR to be approve we have to fix the changes that were out of this scope for this PR. I will review them when there finished.

Once more, thanks for your work. If you have any question related to the project or the review just ask me. The PR is good  and only minor changes have to be made. Thanks for your time :)

File: Backend/docker/build_and_up_dev.sh
[the remark this answers is not in the document]
Comment (xiomaraR) [reply Backend/docker/build_and_up_dev.sh:1]: I think because I had to set execute permission in order to run it

**Comment by xiomaraR** [comment]
@AntonioMrtz thanks for the feedback, I'll try to make the changes suggested ASAP. The import changes for those files you mentioned happened after running the style and linting on the code. I wasn't sure whether to revert it or not at the time

**Comment by xiomaraR** [comment]
I made your suggested changes. I used the `--no-verify` flag when committing because one of the pre-commit hooks altered the imports again, despite my efforts to revert them to their original state. This was the only solution I could think of. Let me know if I should have done anything differently!

**Comment by AntonioMrtz** [comment]
> @AntonioMrtz I made your suggested changes. I used the `--no-verify` flag when committing because one of the pre-commit hooks altered the imports again, despite my efforts to revert them to their original state. This was the only solution I could think of. Let me know if I should have done anything differently!

Hi, using --no-verify flag its a good solution. I will try to have the refactored version ASAP. This will make the overall development easier.

File: Backend/app/routers/login.py (hunk 15 above)
Comment (AntonioMrtz) [reply Backend/app/routers/login.py:2]: Hey, i think this was not reverted. Once its reverted i will run the tests and merge :)

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]
Rollback UTC changes :)

**Comment by AntonioMrtz** [comment]
Once youre finished, can you update the PR description with the final changes? Thanks in advance.

**Review by AntonioMrtz — APPROVED** [review]
Thanks for contributing. Good PR, i hope it helped you understand how is our workflow and motivates you to keep contributing to the project. There are some issues both for backend and frontend and when i have time i will publish more that i have stored on Trello. I hope your contributing experience was great. If it seems good to you i would be eager to see more of your work in the project. This [issue](https://github.com/AntonioMrtz/SpotifyElectron/issues/136) shouldnt take long, take a look if u want. Thanks again and i hope to see you working on the project soon, I will add your github profile to the contributors lists. Regards :)

**Pull request merged**